src/Microsoft.DotNet.Build.Tasks/GatherAssemblyVersionsFromPackageDrop.cs

```diff
@@ -0,0 +1,50 @@
+﻿using Microsoft.Build.Utilities;
+using Microsoft.Build.Framework;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    public class GatherAssemblyVersionsFromPackageDrop : Task
+    {
+        [Required]
+        public string [] PackageDrops { get; set; }
+        [Output]
+        public ITaskItem[] AssemblyNames { get; set; }
+
+        public override bool Execute()
+        {
+            Regex packageNameRegEx = new Regex(@"([^\\]+)\.(\d+\.\d+\.\d+)-([^\.]+)");
+            List<ITaskItem> assemblyNameItems = new List<ITaskItem>();
+
+            foreach (string packageDrop in PackageDrops)
+            {
+                if (!Directory.Exists(packageDrop))
+                {
+                    Log.LogWarning("PackageDrop does not exist - '{0}'", packageDrop);
+                    continue;
+                }
+                IEnumerable<ITaskItem> packages = Directory.GetFiles(packageDrop)?.Select(f => new TaskItem(f));
+
+
+                foreach (ITaskItem package in packages)
+                {
+                    Match m = packageNameRegEx.Match(package.ItemSpec);
+                    if (m.Success)
+                    {
+                        TaskItem assemblyName = new TaskItem(m.Groups[0].Value);
+                        assemblyName.SetMetadata("AssemblyName", m.Groups[1].Value);
```

Comment:
This is making the assumption that the assembly name matches the package name, which is true for most of our library packages, it isn't true for general packages so we might need to think about this more later.